docs/technical-reference/api/moduleConfig-cr.md

```diff
@@ -0,0 +1,97 @@
+# ModuleConfig Custom Resource
+
+The [ModuleConfig custom resource (CR)](../../../api/v1alpha1/moduleconfig_types.go) facilitates the synchronization of `Module` CR configuration between the SKR and KCP clusters, reflecting dynamic changes made by the user.
+
+## Generation
+
+The SKR Watcher notifies the Lifecycle Manager (LM) in case of changes detected in a `Module`. A `ModuleConfig` is then generated/updated by the LM in from the `Module` within the SKR cluster.
+
+## Preconditions
+
+For a `ModuleConfig` to be generated, the following preconditions must be met:
+- The SKR Watcher must be enabled.
+- The `.spec.enableModuleConfig` field in the `ModuleTemplate` must be true. Use the `--enableModuleConfig` CLI flag when generating `ModuleTemplate` to set this field.
+
+## Workflow
+
+The workflow for the `ModuleConfig` involves several steps, orchestrated primarily by the LM and the SKR Watcher:
+
+1. An existing `Module` CR within the SKR is updated by the end-user.
+
+2. The SKR Watcher gets triggered.
+
+3. The SKR Watcher notifies the LM about a change in the `Module` CR.
+
+4. The LM fetches the latest version of `Module` CR from the SKR cluster.
+
+5. Subsequently, the LM generates or updates the `ModuleConfig` from the `Module` within the SKR cluster.
```

Comment:
```suggestion
4. Lifecycle Manager generates or updates a ModuleConfig CR from the Module CR in the Kyma runtime.
```
Using steps is enough to tell that it happens subsequently. We don't need to stress that.